src/elm/App.elm

```diff
@@ -122,6 +123,22 @@ update action model =
         , effects'
         )
 
+    DeleteDigit ->
+      let
+        pincode' =
```

Comment:
Eventhough the button might be disabled, you need to think of this action in API terms, so what happens if it's called without any pins? Here's the answer:

```elm
let
  pincodeLength = 
    length model.pincode

  pincode =
    if length > 0 then String.slice 0 (pincodeLength - 1) model.pincode else ""
in
```